arch/arm/core/aarch32/irq_manage.c

```diff
@@ -74,7 +74,17 @@ void z_arm_irq_priority_set(unsigned int irq, unsigned int prio, uint32_t flags)
 	 * via flags
 	 */
 	if (IS_ENABLED(CONFIG_ZERO_LATENCY_IRQS) && (flags & IRQ_ZERO_LATENCY)) {
+#if defined(CONFIG_ZERO_LATENCY_LEVELS)
+		__ASSERT(CONFIG_ZERO_LATENCY_LEVELS > 0,
+			 "Invalid CONFIG_ZERO_LATENCY_LEVELS %d."
+			 " Value must be > 0.",
+			 CONFIG_ZERO_LATENCY_LEVELS);
+		if (CONFIG_ZERO_LATENCY_LEVELS == 1) {
+			prio = _EXC_ZERO_LATENCY_IRQS_PRIO;
+		}
+#else
 		prio = _EXC_ZERO_LATENCY_IRQS_PRIO;
+#endif
```

Comment:
```suggestion
		if (CONFIG_ZERO_LATENCY_LEVELS == 1) {
			prio = _EXC_ZERO_LATENCY_IRQS_PRIO;
		}
```

Reply:
If the `#if define(CONFIG_ZERO_LATENCY_LEVELS)` is omitted build will fail on targets when CONFIG_ZERO_LATENCY_IRQS is not defined (e.g. nucleo_g0731rb).